### PR TITLE
Feature/modernize tree sitter

### PR DESCRIPTION
**Pull request by mauricioszabo**
This is a feature-branch to track modernization of tree-sitter on Pulsar.

This branch basically should be considered "alpha" for now, and this Draft PR just exist so that we can link binaries on Cirrus and see which tests are failing

Another thing is that this feature is necessary for newer Electron versions (14+)

### CODE REVIEW

File: src/grammar-registry.js [hunk 1]
```diff
@@ -224,14 +227,18 @@ module.exports = class GrammarRegistry {
...
-    this.forEachGrammar(grammar => {
-      const score = this.getGrammarScore(grammar, filePath, fileContents);
-      if (score > highestScore || bestMatch == null) {
-        bestMatch = grammar;
-        highestScore = score;
-      }
-    });
-    return { grammar: bestMatch, score: highestScore };
+    if(this.config.get('core.languageParser') === 'wasm-tree-sitter') {
+      return {grammar: "tree-sitter", score: 10}
```
Comment (icecream17) [inline src/grammar-registry.js:231]: <details><summary>note to self</summary>

ik this is alpha but doesn't this break grammars that don't support the new tree-sitter yet? So really the lines to modify would be 251-259 prev / 258-266 now.

```js
const isNewTreeSitter = grammar instanceof WASMTreeSitterGrammar
// ...
```
oh that doesn't exist. That answers my question.

</details>

NB: `shouldUseOldTreeSitterParser` is just `return false` rn which makes the old tree-sitter tests fail bc it causes text-mate to be preferred over old tree-sitter. (So all the language packages are basically passing.) I assume this is intentional rn.

Comment (mauricioszabo) [reply src/grammar-registry.js:231]: It was, but I'll change it. I had to somehow make it work to at least see the results on my editor, without needing to wire up everything.

As soon as I can, I'll migrate this to the `Grammar` thing and make all configs inside the current `language-*` packages

**Comment by confused-Techie** [comment]
Should've asked much earlier. With the `tree-sitter-hell` variant of tree sitter modernization branches merged, is this draft PR still needed? Or can we close it out?

**Comment by savetheclocktower** [comment]
Closing it out would be my vote, but it's up to @mauricioszabo.

**Comment by mauricioszabo** [comment]
Forgot about this, yes, it's already done :)

**Pull request closed without merging**